### PR TITLE
steps to fix statefulsets in crashloopbackoff

### PR DESCRIPTION
**Pull request by davejrt**

### CODE REVIEW

File: handbook/engineering/on_call/index.md [hunk 1]
```diff
@@ -138,6 +138,29 @@ watch -n1 kubectl get all -n prod -l app=gitserver -o wide
...
+
+#### 1. Update the statefulset `yaml` with the appropriate change and apply using:
+
+```console
+`kubectl apply -f <service.StatefulSet.yaml>`
+```
+#### 2. Delete the pods in the stateful set:
+
+```console
+REPLICAS=`kubectl get sts -n prod <statefulset> -o jsonpath={.spec.replicas}`; for i in `seq 0 $[REPLICAS-1]`; do POD=<statefulset>-$i;   echo "Deleting POD $POD";   kubectl delete pod -n prod $POD ; done
```
[the remark this answers is not in the document]
Comment (davejrt) [reply handbook/engineering/on_call/index.md:160]: what output did this give you? unchanged or patched? I'd be curious to know why this wasn't default 0 

https://cloud.google.com/kubernetes-engine/docs/how-to/updating-apps#partitioning_a_rollingupdate

[the remark this answers is not in the document]
Comment (davejrt) [reply handbook/engineering/on_call/index.md:160]: if we add this into our manifests we can prevent any incident regarding the partitioning happening here:

```
  updateStrategy:
    type: RollingUpdate
    rollingUpdate: 
      partition: 0
```

[the remark this answers is not in the document]
Comment (mrnugget) [reply handbook/engineering/on_call/index.md:160]: > what output did this give you? unchanged or patched? I'd be curious to know why this wasn't default 0

Sorry, I can't remember :(

I have to also add that I'm a huge k8s dilettante and at that point was just re-deleting pods again and again, with the hope of something changing. At some point I noticed that `gitserver-4` _was_ updated, but not `gitserver-{0,1,2,3}`. So I quickly scanned the Kubernetes docs ([here](https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#phased-roll-outs)) and found my exact problem being described (image not changed, even though we updated), so I tried it and it seemed to work.

[the remark this answers is not in the document]
Comment (davejrt) [reply handbook/engineering/on_call/index.md:160]: I get it...the heat of the moment and all that :-) . I personally like the idea of defining this explicitly in our manifests and it being one less thing to worry about when someone gets alerted again.

**Pull request merged**